app/lib/actions/katello/repository/clone_to_version.rb

```diff
@@ -14,40 +14,30 @@ def plan(repositories, content_view_version, options = {})
                                                                version: content_view_version)
 
           rpm_filenames = extract_rpm_filenames(options.fetch(:repos_units, nil), repositories.first.label)
+          fail _('Cannot publish a composite with rpm filenames') if content_view.composite? && rpm_filenames.any?
```

Comment:
is this case even possible ?

Reply:
We block it in the controller here: https://github.com/Katello/katello/blob/master/app/controllers/katello/api/v2/content_views_controller.rb#L88. You can export/import composites, but hammer has some logic where it handles each CVV and then export/imports the composite definition itself.

I'd be OK with leaving or removing this line. Either way, good catch! :fishing_pole_and_fish: :tropical_fish: :fish: :shark:

Reply:
yeah, a controller block is good, i think an action catch is a good idea too, as it provides another layer in the future to catch bad situations and indicates to the developer that they shouldn't have to worry about this situation later on in the action